io/eolearn/io/processing_api.py

```diff
@@ -6,84 +6,51 @@
 import numpy as np
 
 from sentinelhub import WebFeatureService, MimeType, SentinelHubDownloadClient, DownloadRequest, SHConfig,\
-    bbox_to_dimensions
+    bbox_to_dimensions, parse_time_interval, DataSource
 import sentinelhub.sentinelhub_request as shr
 
 from eolearn.core import EOPatch, EOTask, FeatureType
 
 LOGGER = logging.getLogger(__name__)
 
 
-class SentinelHubInputTask(EOTask):
-    """ A processing API input task that loads 16bit integer data and converts it to a 32bit float feature.
-    """
-    def __init__(self, data_source, size=None, resolution=None, bands_feature=None, bands=None, additional_data=None,
-                 maxcc=1.0, time_difference=None, cache_folder=None, max_threads=None, config=None,
-                 bands_dtype=np.float32):
+class SentinelHubInputBase(EOTask):
+    ''' Base class for Processing API input tasks
+    '''
+    def __init__(self, data_source, size=None, resolution=None, cache_folder=None, config=None, max_threads=None):
         """
         :param data_source: Source of requested satellite data.
         :type data_source: DataSource
         :param size: Number of pixels in x and y dimension.
         :type size: tuple(int, int)
         :type resolution: Resolution in meters, passed as a tuple for X and Y axis.
         :type resolution: tuple(int, int)
-        :param bands_feature: Target feature into which to save the downloaded images.
-        :type bands_feature: tuple(sentinelhub.FeatureType, str)
-        :param bands: An array of band names.
-        :type bands: list[str]
-        :param additional_data: A list of additional data to be downloaded, such as SCL, SNW, dataMask, etc.
-        :type additional_data: list[tuple(sentinelhub.FeatureType, str)]
-        :param maxcc: Maximum cloud coverage.
-        :type maxcc: float
-        :param time_difference: Minimum allowed time difference, used when filtering dates, None by default.
-        :type time_difference: datetime.timedelta
         :param cache_folder: Path to cache_folder. If set to None (default) requests will not be cached.
         :type cache_folder: str
+        :param config: An instance of SHConfig defining the service
+        :type config: SHConfig or None
         :param max_threads: Maximum threads to be used when downloading data.
         :type max_threads: int
         """
+
         self.size = size
         self.resolution = resolution
```

Comment:
Maybe the following check could be added:
```Python
if (self.size is None) is (self.resolution is None):
    raise ValueError("Exactly one of the parameters 'size' and 'resolution' should be given, not both")
```
Otherwise cases where none or both are set would be poorly handled.

Reply:
I agree.